diff --git a/views_autocomplete_filters_handler_filter_string.inc b/views_autocomplete_filters_handler_filter_string.inc
index d375cb3..424392d 100644
--- a/views_autocomplete_filters_handler_filter_string.inc
+++ b/views_autocomplete_filters_handler_filter_string.inc
@@ -116,18 +116,33 @@ class views_autocomplete_filters_handler_filter_string extends views_handler_fil
     }
   }

+  /**
+   * Validate that this filter instance has a corresponding autocomplete results field
+   *
+   * @return array    An array of errors triggered by this validation
+   */
   public function validate() {
     $errors = parent::validate();

+    // Only check if this filter instance is exposed and has the autocomplete box checked
     if (!empty($this->options['exposed']) && !empty($this->options['expose']['autocomplete_filter'])) {
-      $fields = $this->view->display_handler->get_handlers('field');
-      $field_name = !empty($this->options['expose']['autocomplete_field']) ? $this->options['expose']['autocomplete_field'] : $this->options['id'];
-      if (empty($field_name) || empty($fields[$field_name])) {
-        $current_display = $this->view->display[$this->view->current_display];
-        $errors[] = t('Field with autocomplete results is not selected for %label filter in %display display.', array(
-          '%label' => $this->ui_name(TRUE),
-          '%display' => $current_display->display_title . ' (' . $this->view->current_display . ')',
-        ));
+      // Look through the view and find the display that this filter instance is part of
+      foreach ($this->view->display as $display_id => $display) {
+        $display_filters = $display->handler->get_handlers('filter');
+        foreach ($display_filters as $filter_id => $filter) {
+          if ($filter === $this) {
+            // At this point we know we're looking at the correct display, and can check the fields
+            // Get the fields from the correct display
+            $display_fields = $display->handler->get_handlers('field');
+            $field_name = !empty($this->options['expose']['autocomplete_field']) ? $this->options['expose']['autocomplete_field'] : $this->options['id'];
+            if (empty($field_name) || empty($display_fields[$field_name])) {
+              $errors[] = t('Field with autocomplete results is not selected for %label filter in %display display.', array(
+                '%label' => $this->ui_name(TRUE),
+                '%display' => $display->display_title . ' (' . $display->id . ')',
+              ));
+            }
+          }
+        }
       }
     }

